lib/types.ts

```diff
@@ -424,7 +424,13 @@ export type Opaque<Type, Token extends string> = Token extends StringLiteral<Tok
   : never;
 
 /** Easily extract the type of a given object's values */
-export type ValueOf<T> = T[keyof T];
+export type ValueOf<T> = T extends Primitive
```

Comment:
Please add changelog with `yarn changeset`

I imagine it will be patch as it's fixing existing utility function

Reply:
Makes sense, added the changeset summary and will update the PR